candi/bashible/common-steps/node-group/064_configure_kubelet.sh.tpl

```diff
@@ -24,44 +24,41 @@ fi
 # This folder doesn't have time to create before we stop freshly, unconfigured kubelet during bootstrap (step 034_install_kubelet_and_his_friends.sh).
 mkdir -p /var/lib/kubelet
 
-# Check CRI type and set appropriated parameters.
-# cgroup default is `systemd`, only for docker cri we use `cgroupfs`.
-cgroup_driver="systemd"
-{{- if eq .cri "Containerd" }}
-# Overriding cgroup type from external config file
-if [ -f /var/lib/bashible/cgroup_config ]; then
-  cgroup_driver="$(cat /var/lib/bashible/cgroup_config)"
-fi
-{{- end }}
-
 {{- if eq .cri "NotManaged" }}
   {{- if .nodeGroup.cri.notManaged.criSocketPath }}
 cri_socket_path={{ .nodeGroup.cri.notManaged.criSocketPath | quote }}
   {{- else }}
-for socket_path in /var/run/docker.sock /run/containerd/containerd.sock; do
-  if [[ -S "${socket_path}" ]]; then
-    cri_socket_path="${socket_path}"
-    break
-  fi
-done
+    {{- if semverCompare "<1.24" .kubernetesVersion }}
+if [[ -S /var/run/docker.sock ]]; then
+  cri_socket_path="${socket_path}"
+  break
+fi
+    {{- end }}
+if [[ -S /run/containerd/containerd.sock ]]; then
+  cri_socket_path="${socket_path}"
+  break
+fi
   {{- end }}
 
 if [[ -z "${cri_socket_path}" ]]; then
   bb-log-error 'CRI socket is not found, need to manually set "nodeGroup.cri.notManaged.criSocketPath"'
   exit 1
 fi
 
+  {{- if semverCompare "<1.24" .kubernetesVersion }}
 if grep -q "docker" <<< "${cri_socket_path}"; then
   cri_type="NotManagedDocker"
 else
   cri_type="NotManagedContainerd"
 fi
-{{- else if eq .cri "Docker" }}
-cri_type="Docker"
+  {{- else }}
+  cri_type="NotManagedContainerd"
+  {{- end }}
 {{- else }}
 cri_type="Containerd"
 {{- end }}
 
+{{- if semverCompare "<1.24" .kubernetesVersion }}
 if [[ "${cri_type}" == "Docker" || "${cri_type}" == "NotManagedDocker" ]]; then
   cgroup_driver="cgroupfs"
```

Comment:
in another cases cgroup_driver is empty. It is an error.

Reply:
Done